### PR TITLE
Add multiline test for rabbitmq

### PR DESCRIPTION
**Pull request by ruflin**
See https://github.com/elastic/beats/issues/902

### CODE REVIEW

**Comment by grumpydude** [comment]
This should be retitled as rabbitmq.

File: filebeat/tests/system/test_multiline.py [hunk 1]
```diff
@@ -71,6 +71,38 @@ def test_c_style_log(self):
...
+        """
+        Test an nginx multiline log
+        Special about this log file is that it has empty new lines
+        """
+        self.render_config_template(
+                path=os.path.abspath(self.working_dir) + "/log/*",
+                multiline=True,
+                pattern="^=[A-Z]+",
+                match="after",
+                negate="true",
```
Comment (urso) [inline filebeat/tests/system/test_multiline.py]: do we want to include the newlines?

Comment (urso) [inline filebeat/tests/system/test_multiline.py]: For example see this playground: http://play.golang.org/p/ABEJaX_lsK

Using the regex `^=[A-Z]+|^$'`, all empty lines become line events of length 0. These empty events should be quite easy filterable via exclude_lines

Comment (urso) [inline filebeat/tests/system/test_multiline.py]: [This pattern](http://play.golang.org/p/6LryE2-n7O) (`^.+`) without negate + `match = before` will include the last empty line. All following empty lines become empty-line-events which can be filtered via exclude_lines.

Comment (ruflin) [inline filebeat/tests/system/test_multiline.py]: About the newlines: I would say it is not the job of filebeat to cleanup log files, means not sending empty lines (keep magic to a minimum).

I much prefer the solution you proposed which makes it possible to handle the problem through configuration and not changes in the logic. We should probably add these examples to our regexp docs (@dede)

**Comment by ruflin** [comment]
@urso I updated the test with a new name and move the content inside the python script. This tests is mainly intended to test the current behaviour and make sure, we detect it in case it changes.

File: filebeat/tests/system/test_multiline.py [hunk 2]
```diff
@@ -71,6 +71,54 @@ def test_c_style_log(self):
         # Check that output file has the same number of lines as the log file
         assert 4 == len(output)
 
+    def test_rabbitmq_multiline_log(self):
+        """
+        Test an nginx multiline log
```
Comment (andrewkroh) [inline filebeat/tests/system/test_multiline.py]: The test description is still wrong. `s/nginx/rabbitmq/`

**Comment by andrewkroh** [comment]
This test case is causing Filebeat to crash. See the log: http://pastebin.com/raw/Lmk8Jwc5

File: filebeat/tests/system/test_multiline.py (hunk 2 above)
Comment (ruflin) [inline filebeat/tests/system/test_multiline.py]: changed

**Comment by ruflin** [comment]
It was actually a problem in the test. So the test runs without panic (I fixed it know). But it revealed the problem that currently Filebeat does not shut down properly when filebeat is stopped in the middle during processing log lines / waiting for completion of a log line. I opened a separate issue to test and fix this: https://github.com/elastic/beats/issues/963

**Pull request merged**